Mathlib/Data/Nat/Prime.lean

```diff
@@ -455,6 +455,21 @@ theorem exists_dvd_of_not_prime2 {n : ℕ} (n2 : 2 ≤ n) (np : ¬Prime n) :
     (not_prime_iff_minFac_lt n2).1 np⟩
 #align nat.exists_dvd_of_not_prime2 Nat.exists_dvd_of_not_prime2
 
+theorem not_prime_of_dvd (h : ∃ m, m ∣ n ∧ m ≠ 1 ∧ m ≠ n) : ¬Prime n := by
+  contrapose! h
+  simp only [← or_iff_not_imp_left]
+  exact h.eq_one_or_self_of_dvd
+
+theorem not_prime_of_dvd2 (h : ∃ m, m ∣ n ∧ 2 ≤ m ∧ m < n) : ¬Prime n := by
+  obtain ⟨m, dvd, hm2, hm⟩ := h
+  exact not_prime_of_dvd ⟨m, dvd, ne_of_gt hm2, ne_of_lt hm⟩
+
+theorem exists_dvd_iff_not_prime {n : ℕ} (n2 : 2 ≤ n) : (¬Prime n) ↔ ∃ m, m ∣ n ∧ m ≠ 1 ∧ m ≠ n :=
+  ⟨exists_dvd_of_not_prime n2, not_prime_of_dvd n2⟩
+
+theorem exists_dvd_iff_not_prime2 {n : ℕ} (n2 : 2 ≤ n) : (¬Prime n) ↔ ∃ m, m ∣ n ∧ 2 ≤ m ∧ m < n :=
+  ⟨exists_dvd_of_not_prime2 n2, not_prime_of_dvd2 n2⟩
```

Comment:
```suggestion
  ⟨exists_dvd_of_not_prime2 n2, not_prime_of_dvd2⟩
```

Reply:
Please resolve old conversations: it makes it easier for later reviewers.